chia/full_node/block_store.py

```diff
@@ -69,7 +69,7 @@ async def add_full_block(self, block: FullBlock, block_record: BlockRecord) -> N
         if cached is not None:
             # Since write to db can fail, we remove from cache here to avoid potential inconsistency
```

Comment:
if the block is cached shouldn't it already be in the database ?
why overwrite and not skip ?

Reply:
There's edge case where we overwrite the same block, iirc related to either compact vdf or finished/unfinished